backend/wallet/services/offchain.py

```diff
@@ -0,0 +1,244 @@
+    # when pass evaluation, we send kyc data as receiver or ready for settlement as sender
+    if command.is_receiver():
+        return _send_kyc_data_and_receipient_signature(command)
+    return command.new_command(status=offchain.Status.ready_for_settlement)
+
+
+def _send_kyc_data_and_receipient_signature(
+    command: offchain.PaymentCommand,
+) -> offchain.PaymentCommand:
+    sig_msg = command.travel_rule_metadata_signature_message(_hrp())
+    user_id = get_account_id_from_subaddr(command.receiver_subaddress(_hrp()).hex())
+
+    return command.new_command(
+        recipient_signature=_compliance_private_key().sign(sig_msg).hex(),
+        kyc_data=_user_kyc_data(user_id),
+        status=offchain.Status.ready_for_settlement,
+    )
+
+
+def _lock_and_save_inbound_command(command: offchain.PaymentCommand) -> Transaction:
+    def validate_and_save(txn: Optional[Transaction]) -> Transaction:
+        if txn:
+            prior = _txn_payment_command(txn)
+            if command == prior:
+                return
+            command.validate(prior)
+            txn.command_json = offchain.to_json(command)
+            txn.status = _command_transaction_status(
+                command, TransactionStatus.OFF_CHAIN_INBOUND
+            )
+        else:
+            txn = _new_payment_command_transaction(
+                command, TransactionStatus.OFF_CHAIN_INBOUND
+            )
+        return txn
+
+    return lock_for_update(command.reference_id(), validate_and_save)
+
+
+def _command_transaction_status(
+    command: offchain.PaymentCommand, default: TransactionStatus
+) -> TransactionStatus:
+    if command.is_both_ready():
+        return TransactionStatus.OFF_CHAIN_READY
+    elif command.is_abort():
+        return TransactionStatus.CANCELED
+    return default
+
+
+def _new_payment_command_transaction(
+    command: offchain.PaymentCommand, status: TransactionStatus
+) -> Transaction:
+    payment = command.payment
+    sender_address, source_subaddress = _account_addresses(payment.sender.address)
+    destination_address, destination_subaddress = _account_addresses(
+        payment.receiver.address
+    )
+    source_id = get_account_id_from_subaddr(source_subaddress)
+    destination_id = get_account_id_from_subaddr(destination_subaddress)
+
+    return Transaction(
+        type=TransactionType.OFFCHAIN,
+        status=status,
+        amount=payment.action.amount,
+        currency=payment.action.currency,
+        created_timestamp=datetime.utcnow(),
+        source_id=source_id,
+        source_address=sender_address,
+        source_subaddress=source_subaddress,
+        destination_id=destination_id,
+        destination_address=destination_address,
+        destination_subaddress=destination_subaddress,
+        reference_id=command.reference_id(),
+        command_json=offchain.to_json(command),
+    )
+
+
+def _account_addresses(account_id: str) -> Tuple[str, Optional[str]]:
```

Comment:
nit: _account_address

Reply:
this function returns account_address and sub_address, hence it is named with "es", use "account_address" probably is not good, so I will rename it to account_address_and_subaddress